rewriter/Struct.cc

```diff
@@ -112,14 +112,32 @@ vector<ast::ExpressionPtr> Struct::run(core::MutableContext ctx, ast::Assign *as
         }
     }
 
+    UnorderedSet<core::NameRef> seenNames;
     for (auto &arg : send->posArgs()) {
-        auto sym = ast::cast_tree<ast::Literal>(arg);
-        if (!sym || !sym->isSymbol()) {
+        auto lit = ast::cast_tree<ast::Literal>(arg);
+        if (!lit) {
             return empty;
         }
-        core::NameRef name = sym->asSymbol();
-        auto symLoc = sym->loc;
-        auto strname = name.shortName(ctx);
+
+        std::string strname;
+        if (lit->isSymbol()) {
+            strname = lit->asSymbol().show(ctx);
+        } else if (lit->isString()) {
+            strname = lit->asString().show(ctx);
+        } else {
+            return empty;
+        }
+
+        core::NameRef name = ctx.state.enterNameUTF8(strname);
```

Comment:
There should be no need to do this conversion from `NameRef` to string and then back to `NameRef`: `asSymbol` and `asString` already return `NameRef`s, and a symbol `:foo` and a string `"foo"` are guaranteed to have the same underlying `NameRef`.  So you can use the `NameRef` of the literal directly:

```c++
core::NameRef name;
if (lit->isSymbol()) {
    name = lit->asSymbol();
} else if (lit->isString()) {
    name = lit->asString();
} else {
    return empty;
}
```